internal/broker/broker.go

```diff
@@ -735,3 +840,40 @@ func decorateErrorMessage(data *isAuthenticatedDataResponse, msg string) {
 	errMsg.Message = fmt.Sprintf("%s: %s", msg, errMsg.Message)
 	*data = errMsg
 }
+
+func cleanupOldEncryptedToken(path string) {
+	if err := os.Remove(path); err != nil {
+		slog.Warn(fmt.Sprintf("Failed to remove old encrypted token %s: %v", path, err))
+		return
+	}
+
+	// Also remove the parent directory and the parent's parent directory if they are empty. The directory structure was:
+	//   $SNAP_DATA/cache/$ISSUER/$USERNAME.cache
```

Comment:
Where do you think such a comment would make sense? I'm hesitant, because I expect that such a comment is either in a place where it doesn't benefit us much, because it's easy to figure out the path by reading the surrounding code (like https://github.com/ubuntu/authd-oidc-brokers/blob/369-store-token-unencrypted/internal/broker/broker.go#L184-L184) or in a place which we're likely to forget to update if we ever change the path (because it's not close to where the path is defined).

Reply:
The part you hilighted only define the base path. What I thought in particular was on the latter part (the `$ISSUER/$USERNAQME.cache`) or rather its replacement, to where the code lives. Does it make sense?

Reply:
> What I thought in particular was on the latter part (the $ISSUER/$USERNAQME.cache) or rather its replacement

That's defined just above, in https://github.com/ubuntu/authd-oidc-brokers/blob/0309d926a9304614b1ab91dec513f811306fc40b/internal/broker/broker.go#L179-L182

I added comments above the assignments to `session.tokenPath` and `session.passwordPath`.